src/Functions/tupleElement.cpp

```diff
@@ -113,9 +113,14 @@ class FunctionTupleElement : public IFunction
 private:
     size_t getElementNum(const ColumnPtr & index_column, const DataTypeTuple & tuple) const
     {
-        if (const auto * index_col = checkAndGetColumnConst<ColumnUInt8>(index_column.get()))
+        if (
+            checkAndGetColumnConst<ColumnUInt8>(index_column.get())
+                || checkAndGetColumnConst<ColumnUInt16>(index_column.get())
+                || checkAndGetColumnConst<ColumnUInt32>(index_column.get())
+                || checkAndGetColumnConst<ColumnUInt64>(index_column.get())
+        )
         {
-            size_t index = index_col->getValue<UInt8>();
+            size_t index = index_column->getUInt(0);
```

Comment:
Wanted to add a test, that for given arch we can handle max UInt64, but could not find the way to construct automatically very long Tuple. Made a feature request #10882